include/picongpu/particles/atomicPhysics/ionizationPotentialDepression/StewartPyattIPD.hpp

```diff
@@ -255,20 +272,57 @@ namespace picongpu::particles::atomicPhysics::ionizationPotentialDepression
             // UNIT_ENERGY/eV
             constexpr float_X eV = sim.pic.get_eV();
 
+            float_X const chargeState_X = static_cast<float_X>(chargeState);
+
             // UNIT_MASS * UNIT_LENGTH^3 / UNIT_TIME^2 * 1/(eV * UNIT_ENERGY/eV * UNIT_LENGTH)
             // = unitless, not weighted
             float_X const K
                 = (pmacc::math::isApproxZero(
                       superCellConstantInput.temperatureTimesk_Boltzman * superCellConstantInput.debyeLength))
                       ? 0._X
-                      : constFactor * (chargeState + 1)
+                      : constFactor * (chargeState_X + 1._X)
                             / (superCellConstantInput.temperatureTimesk_Boltzman * eV
                                * superCellConstantInput.debyeLength);
 
             // eV, not weighted
-            return superCellConstantInput.temperatureTimesk_Boltzman
-                   * (math::pow((3 * (superCellConstantInput.zStar + 1) * K + 1), 2._X / 3._X) - 1._X)
-                   / (2._X * (superCellConstantInput.zStar + 1._X));
+            float_X const stewartPyattIPD
+                = superCellConstantInput.temperatureTimesk_Boltzman
+                  * (math::pow((3._X * (superCellConstantInput.zStar + 1._X) * K + 1._X), 2._X / 3._X) - 1._X)
+                  / (2._X * (superCellConstantInput.zStar + 1._X));
+
+            float_X result = 0._X;
+
+            // additional ipd contribution as implemented in scfly
+            if constexpr(T_useSCFLYextendedStewartPyattIPD)
+            {
+                /** @details taken directly from the scfly source code,
+                 *  https://github.com/ComputationalRadiationPhysics/scfly/blob/5d9b0a997bb6688b04e1a34a7fa8db0f2657106a/code/src/scdrv.f#L3519-L3547
+                 *
+                 * no source known, Brian Marre, 2025
+                 */
+                // (1/(1/unit_length^3))^(1/3) = unit_length
+                float_X const ionSphereRadius = math::pow(
+                    0.75_X * (chargeState_X + 1._X) / (picongpu::PI * superCellConstantInput.freeElectronDensity),
+                    1. / 3.);
+
+                // (eV * cm) * m/cm * unit_length/m = eV * unit_length
+                constexpr float_X constFactorSCFLY = 1.45e-7_X / sim.unit.length() * 1.e-2_X;
+
+                // 1 * eV * unit_length / sqrt(unit_length^2 + unit_length^2) = eV, not weighted
+                float_X const scfly_secondBranch
+                    = (chargeState_X + 1._X) * constFactorSCFLY
+                      / math::sqrt(
+                          (ionSphereRadius / 1.5_X) * (ionSphereRadius / 1.5_X)
+                          + (superCellConstantInput.debyeLength) * (superCellConstantInput.debyeLength));
+
+                result = math::min(scfly_secondBranch, stewartPyattIPD);
```

Comment:
Where is the multiplication with `zz` from the original code?

Reply:
I assume `zz` = `(chargeState_X + 1._X)`. Since both branches include that delay the multiplication to this point here as the original implementation.